contracts/PriceOracle/PriceOracle.sol

```diff
@@ -155,30 +190,59 @@ contract PriceOracle is Ownable2Step {
      */
     function addConfig(LoadConfig memory config) public onlyOwner {
         _validateTokenConfig(config);
-        TokenConfig memory tokenConfig = TokenConfig(config.underlyingAssetDecimals, config.priceFeed);
+        TokenConfig memory tokenConfig = TokenConfig(config.underlyingAssetDecimals, config.priceFeed, config.fixedPrice);
         tokenConfigs[config.cToken] = tokenConfig;
-        emit PriceOracleAssetAdded(config.cToken, config.underlyingAssetDecimals, config.priceFeed);
+        emit PriceOracleAssetAdded(config.cToken, config.underlyingAssetDecimals, config.priceFeed, config.fixedPrice);
     }
 
     /**
-     * @notice Updates the price feed in the token config for a particular cToken
+     * @notice Updates the price feed in the token config for a particular cToken and sets fixed price to 0 if it is not already.
      * @param cToken The cToken address that the config needs to be updated for
      * @param priceFeed The address of the new price feed the config needs to be updated to
      */
     function updateConfigPriceFeed(address cToken, address priceFeed) external onlyOwner {
         TokenConfig memory config = tokenConfigs[cToken];
         // Check if config exists for cToken
-        if (config.priceFeed == address(0)) revert ConfigNotFound(cToken);
+        if (config.underlyingAssetDecimals == 0) revert ConfigNotFound(cToken);
         // Validate price feed
-        if (priceFeed == address(0)) revert InvalidPriceFeed(priceFeed);
+        if (priceFeed == address(0)) revert MissingPriceFeed();
         // Check if existing price feed is the same as the new one sent
         if (config.priceFeed == priceFeed) revert UnchangedPriceFeed(cToken, config.priceFeed, priceFeed);
         // Validate the decimals for the price feed since it could differ from the previous one
         _validateDecimals(priceFeed, config.underlyingAssetDecimals);
 
         address existingPriceFeed = config.priceFeed;
-        tokenConfigs[cToken].priceFeed = priceFeed;
-        emit PriceOracleAssetPriceFeedUpdated(cToken, existingPriceFeed, priceFeed);
+        uint256 existingFixedPrice = config.fixedPrice;
+        TokenConfig storage storageConfig = tokenConfigs[cToken];
+        storageConfig.priceFeed = priceFeed;
+        if (config.fixedPrice != 0) {
+            storageConfig.fixedPrice = 0;
+        }
+        emit PriceOracleAssetPriceFeedUpdated(cToken, existingPriceFeed, priceFeed, existingFixedPrice);
+    }
+
+    /**
+     * @notice Updates the fixed price in the token config for a particular cToken and sets price feed to 0 if it is not already.
+     * @param cToken The cToken address that the config needs to be updated for
+     * @param fixedPrice The fixed price to be returned for the asset. Expected to be formatted.
+     */
+    function updateConfigFixedPrice(address cToken, uint256 fixedPrice) external onlyOwner {
```

Comment:
Is the Compound Community ok that we can update the fixed price whenever we want? The alternative is that we make these fixed prices immutable

Reply:
We'd be transferring ownership of this contract to the Compound multisig so it would be up to them if they'd like to update. This method was added to allow Compound to deprecate assets in the future without needing a redeploy. They could just set a fixed price for an asset during their deprecation process and be unaffected when we decomm the feed.